### PR TITLE
fix(FX-4319): commercial buttons section overflow

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Fix**

This PR solves [FX-4319]

Review app: [commercial-buttons](http://commercial-buttons.artsy.net/)
Quick artwork example: [Link](https://commercial-buttons.artsy.net/artwork/damon-zucconi-editioned-cg-price-range-arta-us-2)

### Problems
* Commercial buttons starts overflowing the sidebar when screen width < 1075
* iPhone SE (first generation) does the same when contact gallery is secondary

### Solution
We discussed with Kseniya and decided that the buttons will be placed under each other by default (column view). When the screen width is `1192px` or more (`xl` breakpoint according to the list of [supported breakpoints](https://palette.artsy.net/guides/responsive/#all-breakpoints)), then place the buttons in a row

#### Problem №1 - when screen width < 1075
| Before | After |
| ------------- | ------------- |
| before-1 | after-1 | 

#### Problem №2 - when the device has a small screen (e.g. iPhone SE first generation)
| Before | After |
| ------------- | ------------- |
| before-2 | after-2 | 

[FX-4319]: https://artsyproduct.atlassian.net/browse/FX-4319?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/Artwork/Components/ArtworkSidebar2/ArtworkSidebar2CommercialButtons.tsx [hunk 1]
```diff
@@ -287,10 +289,50 @@ const ArtworkSidebar2CommerialButtons: React.FC<ArtworkSidebar2CommercialButtons
       return null
     }
 
+    return (
+      <ArtworkSidebarCreateAlertButtonFragmentContainer artwork={artwork} />
+    )
+  }
+
+  const renderButtons = () => {
```
Comment (dzucconi) [inline src/Apps/Artwork/Components/ArtworkSidebar2/ArtworkSidebar2CommercialButtons.tsx:297]: We should really avoid this pattern of having a 'render' function inside of another render function. Ideally should be extracted to its own component. If you really don't want to do that then wrap in `useMemo`. (It's caused subtle bugs in production in the past.)

**Comment by dzucconi** [comment]
(OH, I see it's a feature flag. What about just simply changing "Contact Gallery" to "Contact" at the smaller size with a simple `<span display={["none", "inline"]}>Gallery</span>`

**Comment by dzucconi** [comment]
Also couldn't you just achieve this proposed fix with a responsive prop for `flexDirection`? (I'm just convinced we over rely `Media`)

**Comment by dimatretyak** [comment]
> Also couldn't you just achieve this proposed fix with a responsive prop for `flexDirection`? (I'm just convinced we over rely `Media`)

we can do this, but in this case it will look something [like this in the code](https://github.com/artsy/force/blob/5fcfb8285177c3ad23e43ce88428df63a1331d34/src/Apps/Artwork/Components/ArtworkSidebar2/ArtworkSidebar2CommercialButtons.tsx#L328) (it seemed to me that it was better to use `Media` and `renderButtons` pattern)

**Comment by dzucconi** [comment]
Responsive props are going to be much more light weight and are preferable. I only really reach for `Media` if it's like two totally different designs.

**Comment by dzucconi** [comment]
The use of `Media` for simply a copy change is kind of interesting, I guess. Though I don't see what benefit that has over the `span`

**Comment by dimatretyak** [comment]
> The use of `Media` for simply a copy change is kind of interesting, I guess. Though I don't see what benefit that has over the `span`

I mean, here we use `react-i18next` and it won't work so easy to use `span` with `display={["none", "inline"]}`

**Comment by dimatretyak** [comment]
> What about just simply changing "Contact Gallery" to "Contact" at the smaller size

We discussed this moment with Kseniya and decided to leave the current implementation

File: src/Apps/Artwork/Components/ArtworkSidebar2/ArtworkSidebar2CommercialButtons.tsx [hunk 2]
```diff
@@ -326,21 +325,32 @@ const ArtworkSidebar2CommerialButtons: React.FC<ArtworkSidebar2CommercialButtons
...
+          separator={
+            <>
+              <Media at="xl">
+                <Spacer ml={1} />
+              </Media>
+
+              <Media lessThan="xl">
+                <Spacer mt={1} />
+              </Media>
+            </>
```
Comment (dzucconi) [inline src/Apps/Artwork/Components/ArtworkSidebar2/ArtworkSidebar2CommercialButtons.tsx:339]: These Spacers aren't any different? (And if they were: responsive props)

File: src/Apps/Artwork/Components/ArtworkSidebar2/ArtworkSidebar2CommercialButtons.tsx [hunk 3]
```diff
@@ -326,21 +325,32 @@ const ArtworkSidebar2CommerialButtons: React.FC<ArtworkSidebar2CommercialButtons
...
+        <Join
+          separator={
+            <>
+              <Media at="xl">
+                <Spacer ml={1} />
+              </Media>
+
+              <Media lessThan="xl">
+                <Spacer mt={1} />
+              </Media>
```
Comment (dzucconi) [inline src/Apps/Artwork/Components/ArtworkSidebar2/ArtworkSidebar2CommercialButtons.tsx:338]: `<Spacer ml={[0, 0, 0, 1]} mt={[1, 0, 0, 0]} />` or something.

File: src/Apps/Artwork/Components/ArtworkSidebar2/ArtworkSidebar2CommercialButtons.tsx [hunk 4]
```diff
@@ -326,21 +325,32 @@ const ArtworkSidebar2CommerialButtons: React.FC<ArtworkSidebar2CommercialButtons
...
-            loading={isCommitingCreateOrderMutation}
-            onClick={handleCreateOrder}
-          >
-            {t("artworkPage.sidebar.commercialButtons.buyNow")}
-          </Button>
-        )}
-        {artwork.isOfferable && (
-          <>
-            <Spacer ml={artwork.isAcquireable ? 1 : 0} />
+      <Flex flexDirection={["column", "column", "column", "column", "row"]}>
```
Comment (dzucconi) [inline src/Apps/Artwork/Components/ArtworkSidebar2/ArtworkSidebar2CommercialButtons.tsx:328]: This is probably correct but why wouldn't `["column", "row"]` work here?

File: src/Apps/Artwork/Components/ArtworkSidebar2/ArtworkSidebar2CommercialButtons.tsx [hunk 5]
```diff
@@ -326,21 +324,20 @@ const ArtworkSidebar2CommerialButtons: React.FC<ArtworkSidebar2CommercialButtons
...
+        <Join separator={<Spacer ml={[0, 0, 0, 0, 1]} mt={[1, 1, 1, 1, 0]} />}>
```
Comment (dzucconi) [inline src/Apps/Artwork/Components/ArtworkSidebar2/ArtworkSidebar2CommercialButtons.tsx:328]: And actually just simply `<Spacer mt={1} ml={1} />` would work too! 😛

Comment (dimatretyak) [reply src/Apps/Artwork/Components/ArtworkSidebar2/ArtworkSidebar2CommercialButtons.tsx]: thank you!

File: src/Apps/Artwork/Components/ArtworkSidebar2/ArtworkSidebar2CommercialButtons.tsx (hunk 4 above)
Comment (dzucconi) [reply src/Apps/Artwork/Components/ArtworkSidebar2/ArtworkSidebar2CommercialButtons.tsx:327]: 👍 👍

**Review by dzucconi — APPROVED** [review]

**Pull request merged**